src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/F1R2FilterConstants.java

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.hellbender.tools.walkers.readorientation;
+
+import htsjdk.samtools.util.SequenceUtil;
+import org.broadinstitute.hellbender.utils.Nucleotide;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Created by tsato on 3/14/18.
+ */
+public class F1R2FilterConstants {
+    // Padding on each side
+    public static final int REF_CONTEXT_PADDING = 1;
+    public static final int NUM_STATES = ArtifactState.values().length;
+
+    static final int MIDDLE_INDEX = REF_CONTEXT_PADDING;
+
+    static final int REFERENCE_CONTEXT_SIZE = 2 * REF_CONTEXT_PADDING + 1; // aka 3
+
+    public static final List<Nucleotide> REGULAR_BASES = Arrays.asList(Nucleotide.A, Nucleotide.C, Nucleotide.G, Nucleotide.T);
+
+    // the list of all possible kmers, where k = REFERENCE_CONTEXT_SIZE
+    static final List<String> ALL_KMERS = SequenceUtil.generateAllKmers(REFERENCE_CONTEXT_SIZE).stream()
+            .map(String::new).collect(Collectors.toList());
+
+    // For each pair of a K-mer and its reverse complement, pick lexicographically smaller K-mer
+    // to be the canonical representation of the pair
+    static final List<String> CANONICAL_KMERS = ALL_KMERS.stream()
+            .map(context -> new TreeSet<>(Arrays.asList(context, SequenceUtil.reverseComplement(context))))
+            .distinct()
+            .map(s -> s.first().compareTo(s.last()) < 0 ? s.first() : s.last())
+            .collect(Collectors.toList());
+
+    // Delimiter of the alt histogram label
+    static final String FIELD_SEPARATOR = "_";
+
+    static final String binName = "depth";
+
+    // We combine all sites of depths above this value in the last bin of the histogram
+    static final int maxDepth = 200;
```

Comment:
This should be a command line argument.

Reply:
I agree this should be an argument but it's tricky. Both `CollectF1R2Counts` and `LearnReadOrientationModel` use this variable, and they should be in sync. I could either

1. Make it an argument in both tools and note in comment that if the user changes this value for one tool the user should also change it for the other
2. Make it an argument for `CollectF1R2Counts` only, and write it to the reference metrics histogram file as a metric. Then `LearnReadOrientationModel` will get the number from the file.

Do you prefer one over the other?

Reply:
Chose 1)

Reply:
I'm fine with 1).